x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/bulk_edit_rule_errors_list.test.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { __IntlProvider as IntlProvider } from '@kbn/i18n-react';
+import { render, screen } from '@testing-library/react';
+
+import type { DryRunResult } from './use_bulk_actions_dry_run';
+import { BulkEditRuleErrorsList } from './bulk_edit_rule_errors_list';
+import { BULK_ACTIONS_DRY_RUN_ERR_CODE } from '../../../../../../../common/constants';
+
+const Wrapper: FC = ({ children }) => {
+  return (
+    <IntlProvider locale="en">
+      <>{children}</>
+    </IntlProvider>
+  );
+};
+
+describe('Component BulkEditRuleErrorsList', () => {
+  test('should not render component if no errors present', () => {
+    const { container } = render(<BulkEditRuleErrorsList ruleErrors={[]} />, { wrapper: Wrapper });
+
+    expect(container.childElementCount).toEqual(0);
+  });
+
+  test('should render multiple error messages', () => {
+    const ruleErrors: DryRunResult['ruleErrors'] = [
+      {
+        message: 'test failure',
+        ruleIds: ['rule:1', 'rule:2'],
+      },
+      {
+        message: 'another failure',
+        ruleIds: ['rule:1'],
+      },
+    ];
+    render(<BulkEditRuleErrorsList ruleErrors={ruleErrors} />, { wrapper: Wrapper });
+
+    expect(screen.getByText("2 rules can't be edited (test failure)")).toBeInTheDocument();
+    expect(screen.getByText("1 rule can't be edited (another failure)")).toBeInTheDocument();
+  });
+
+  test.each([
```

Comment:
Let's avoid using `test.each` whenever possible. That method makes tests much harder (I'd say impossible) to debug using breakpoints and does not allow isolating failing tests with `test.only`.

Reply:
I can see it used widely across our app. And I like it as it allows to cover a lot of test cases with a minimal amount of code.

>  does not allow isolating failing tests with test.only.
Possible way to do it: wrap it in describe, and leave a case you would like to test. While it's not ideal of course, that's a trade-off for less code and easier maintainability(imagine if there is a change in test case: you would need to do it N places instead of one)